### DISCUSSION TITLE
Keycloak 12.0.4 with Oauth2-proxy to Authenticate NGINX requests

### DISCUSSION THREAD

**Discussion by cnhsn** [keycloak/keycloak#22462] [Q&A]
Hello everyone,

I use a reverse proxy to access some of my applications with NGINX. I currently protect my reverse proxies with Oauth2-proxy and authorize accessing users with Azure AD. Instead of using Azure AD directly with Oauth2-proxy, I decided to use Keycloak as IdP. I installed Keycloak and it is working fine now. I also added Azure AD as a provider and it works fine. What I want to do now is, I want to use Keycloak as a provider in the oauth2-proxy module. I would also like to define allowed_groups in the oauth2-proxy config file and define a different access group for each reverse proxy and make group-based access management. How can I do that? I'd really appreciate your help.

**Comment by codespearhead** [comment]
Use [oauth2-proxy's Keycloak OIDC provider](https://oauth2-proxy.github.io/oauth2-proxy/configuration/providers/keycloak_oidc).

As for the second question, I suggest asking the [oauth2-proxy community](https://github.com/oauth2-proxy/oauth2-proxy?tab=readme-ov-file#getting-involved).
